content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md

```diff
@@ -0,0 +1,320 @@
+
+In this blog post, we c﻿hoose the Apache SkyWalking as an APM tool and describe how to set up it, as a *self-hosted* APM tool, for monitoring and alerting the application performance across hybrid cloud environment. We take the service mesh as an auto instrumentation mechanism for application monitoring, without adding any manual process to instrument existing applications. 
+
+## Apache SkyWalking
+
+[Apache SkyWalking](https://skywalking.apache.org/) is an open source APM tool with capabilities for monitoring, tracing and diagnosing distributed system. It’s especially designed for microservices, cloud native and container-based architectures. 
+
+Apart from a list of supported agents to be used for instrumenting applications, Apache SkyWalking implements an [Envoy Access Log Service (ALS)](https://www.envoyproxy.io/docs/envoy/v1.18.2/api-v2/service/accesslog/v2/als.proto) based solution to provide observability on the service mesh, no matter the architecture or language. Since service mesh provides full control of *RPC*, this observation solution is much easier to be added without language specific technology. With this solution, users could get the application service topology map, metrics graph, request details and error message with a very nice visualization. This integration solution can be extremely important for monitoring and visualizing application that consists of many microservices running across on-premises, cloud-based or hybrid environments. 
+
+Apache SkyWalking is lightweight and scalable. It can be easily set up as a *self-hosted* APM tool within hybrid cloud environment, without any additional external resource for hosting the tool. This can help in the context of a resource-constrained environment and remove the privacy and security concerns customers may have to put customer data going to third party services. 
+
+## Set up Apache SkyWalking
+
+### Prerequisites
```

Comment:
In the list of prerequisites, you may also want to add the requirement for "istioctl" command line utility installed on the local development environment with hotlink to the istioctl installation: https://istio.io/latest/docs/setup/install/istioctl/

What do you think?

Reply:
Good catch ! yes, both of those CLI tools are required. They were in my original content, strange. I've just re-added them.